File: Taller java/src/nivel_4/Promedio.java
package nivel_4;
import java.util.Scanner;

public class Promedio {
    public static double promedio(double... nums) {
        if (nums.length == 0) throw new IllegalArgumentException("Se requieren al menos un número.");
        double suma = 0;
        for (double v : nums) suma += v;
        return suma / nums.length;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingresa números separados por comas para calcular el promedio: ");
        String entrada = sc.nextLine();
        String[] partes = entrada.split(",");
        double[] nums = new double[partes.length];
        for (int i = 0; i < partes.length; i++) nums[i] = Double.parseDouble(partes[i].trim());
        System.out.println("Promedio: " + promedio(nums));
        sc.close();
    }
}

File: programacion niveles Python/Nivel 4/promedio.py
def promedio(*args):
    if len(args) == 0:
        raise ValueError("Se requieren al menos un número para calcular el promedio.")
    return sum(args) / len(args)

# main
if __name__ == "__main__":
    entrada = input("Ingrese números separados por comas para calcular el promedio: ")
    nums = [float(x.strip()) for x in entrada.split(",") if x.strip() != ""]
    print("Promedio:", promedio(*nums))


